config/application.rb

```diff
@@ -18,6 +18,8 @@
 # you've limited to :test, :development, or :production.
 Bundler.require(*Rails.groups)
 
+require_relative '../lib/types'
```

Comment:
A common paradigm I've seen is to put files like this into `app/lib` instead of `lib`. You're breaking code reloading when you manually require like this. I'd suggest you move the folder and delete this line.

Reply:
Do you have any links to posts about this topic? I'm interested to learn more about this

Reply:
https://guides.rubyonrails.org/autoloading_and_reloading_constants.html#reloading
> Rails automatically reloads classes and modules if application files in the autoload paths change.

If it isn't autoloaded by Zeitwerk, it's not reloaded automatically when you write to the files. That means you need top stop and start `bin/dev` for those changes to take effect.

https://github.com/fxn/zeitwerk/issues/187#issuecomment-946179214
Xavier recommends `app/lib` instead of `lib` for any code that should be autoloadable (so like classes would go in app/lib and rake tasks go in lib).

Reply:
Apologies for the delay, moving `lib/types.rb` into `app/types` is better, thank you for sharing! 🙇🏻

Reply:
No problem!